clang/lib/CConv/Constraints.cpp

```diff
@@ -283,17 +283,58 @@ static bool do_solve(ConstraintsGraph &CG,
   return ok;
 }
 
-auto isNonParam =
-    [](VarAtom *VA) -> bool {
-      VarAtom::VarKind VK = VA->getVarKind();
-      return VK != VarAtom::V_Param;
-    };
-
-auto isNonParamReturn =
-    [](VarAtom *VA) -> bool {
-      VarAtom::VarKind VK = VA->getVarKind();
-      return VK != VarAtom::V_Param && VK != VarAtom::V_Return;
-    };
+VarAtomPred isReturn = [](VarAtom *VA) -> bool {
+  return VA->getVarKind() == VarAtom::V_Return;
+};
+VarAtomPred isParam = [](VarAtom *VA) -> bool {
+  return VA->getVarKind() == VarAtom::V_Param;
+};
+VarAtomPred isNonParamReturn = [](VarAtom *VA) -> bool {
+  return !isReturn(VA) && !isParam(VA);
+};
+
+// Remove from S all elements that don't match the predicate P
+void filter(VarAtomPred P, std::set<VarAtom *> &S) {
+  for (auto I = S.begin(), E = S.end(); I != E; ) {
+    if (!P(*I))
+      I = S.erase(I);
+    else
+      ++I;
+  }
+}
+
+static std::set<VarAtom *> findBounded(ConstraintsGraph &CG, std::set<VarAtom *> *Concrete, bool Succs, bool UseConstAtoms) {
```

Comment:
Can you add a comment header to explain what this function is doing, in terms of its arguments? Also, make sure it fits in 80 columns (following `clang` formatting guidelines, IIRC).